src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -16,8 +16,42 @@ export default class DefaultTransceiverController implements TransceiverControll
 
   constructor(protected logger: Logger, protected browserBehavior: BrowserBehavior) {}
 
-  setEncodingParameters(_params: Map<string, RTCRtpEncodingParameters>): void {
-    return;
+  async setEncodingParameters(encodingParamMap: Map<string, RTCRtpEncodingParameters>): Promise<void> {
+    if (!this._localCameraTransceiver || this._localCameraTransceiver.direction !== 'sendrecv') {
+      return;
+    }
+
+    const sender = this._localCameraTransceiver.sender;
+    const newEncodingParams = Array.from(encodingParamMap.values());
+    if (newEncodingParams.length <= 0) {
+      return;
+    }
+    let hasChanged = false;
+
+    const oldParam: RTCRtpSendParameters = sender.getParameters();
+    if (!oldParam.encodings) {
+      oldParam.encodings = newEncodingParams;
+      hasChanged = true;
+    } else {
+      for (let i = 0; i < oldParam.encodings.length; i++) {
+        for (let j = 0; j < newEncodingParams.length; j++) {
+          if (oldParam.encodings[i].rid === newEncodingParams[j].rid) {
+            if (oldParam.encodings[i].maxBitrate !== newEncodingParams[j].maxBitrate) {
+              oldParam.encodings[i].maxBitrate = newEncodingParams[j].maxBitrate;
+              hasChanged = true;
+            }
+            if (oldParam.encodings[i].scaleResolutionDownBy !== newEncodingParams[j].scaleResolutionDownBy) {
+              oldParam.encodings[i].scaleResolutionDownBy = newEncodingParams[j].scaleResolutionDownBy;
+              hasChanged = true;
+            }
+          }
+        }
+      }
+    }
+
+    if (hasChanged) {
+      await sender.setParameters(oldParam);
+    }
```

Comment:
Is there any reason not to just always call `setParameters`?

Can we just expect the policy to return an empty map if it gets to see `Sender` and its own state, and doesn't think anything needs to change?